### PR TITLE
Build with mingw

### PR DESCRIPTION
**Pull request by jcelaya**
These patches are needed to build noPoll for Windows. Tested with MinGW64 5.3, both on Windows and cross-compiled from Fedora 23.

### CODE REVIEW

File: src/nopoll_decl.h [hunk 1]
```diff
@@ -168,13 +168,17 @@
 /* additional includes for the windows platform */
 
 /* _WIN32_WINNT note: If the application including the header defines
- * the _WIN32_WINNT, it must include the bit defined by the value
- * 0x400. */
+ * the _WIN32_WINNT, it must include the bits defined by the value
+ * 0x501. */
```
Comment (francisbrosnan) [inline src/nopoll_decl.h]: Javier, what's the purpose of this change?

Comment (jcelaya) [inline src/nopoll_decl.h]: In MinGW for Windows, function declarations of getaddrinfo and freeaddrinfo are guarded by an "#if _WIN32_WINNT >= 0x501" in ws2tcpip.h. Surprisingly, they are not guarded in MinGW for Linux.

File: configure.ac [hunk 2]
```diff
@@ -55,7 +55,7 @@ compiler_options=""
 STRICT_PROTOTYPES=""
 echo "Detected compiler: $compiler"
 if test "$compiler" = "gcc" ; then
-   compiler_options="-Wstrict-prototypes -Wall -Werror -ansi -fstack-protector-all -Wstack-protector"
```
Comment (jcelaya) [inline configure.ac]: It's OK for us. Only MinGW for Windows fails to build with "-ansi", it claims that strdup does not exist. We always do cross-compilation from Linux, so you can just skip this patch.

File: src/nopoll_decl.h (hunk 1 above)
Comment (jcelaya) [inline src/nopoll_decl.h]: As I replied in the other review, we always do cross-compilation from Linux. If forcing _WIN32_WINNT to 0x501 bothers you, we do not mind keeping it as 0x400.

**Comment by jcelaya** [comment]
Sorry, but I am not very fluent with PRs. Is there anything I should do next? AFAIK, you could accept only the two approved commits, but I can change the source branch to another one without the ANSI C commit, if you want.

**Comment by francisbrosnan** [comment]
Hello Javier,
Please, go ahead. I have no option to accept this PR without ansi removal,
Thanks Javier,

**Comment by jcelaya** [comment]
Done

**Comment by francisbrosnan** [comment]
Good, thanks for the update and reporting Javier,
Best Regards,
